### ISSUE TITLE
BitBucket integration: bad username attribute name when enabling integration

### ISSUE THREAD

**Issue by trapped** [getsentry/sentry#15487]
## Important Details

How are you running Sentry?

* [x] On-Premise docker [Version 9.1.2]
* [ ] Saas (sentry.io)
* [ ] Other [briefly describe your environment]

## Description

When successfully authorizing the self-hosted Sentry app in BitBucket, a HTTP 500 Internal Server Error is raised and integration fails.

## Steps to Reproduce

1. Deploy self-hosted Sentry 9.1.2 (Docker with https://github.com/getsentry/onpremise)
2. Enable Development mode in BitBucket (for unverified integrations)
3. Try activating BitBucket integration in Sentry, which redirects to the BitBucket OAuth consent screen
4. Authorize, which will fail with `500 Internal Server Error`
5. Check exception in Sentry `internal` project's Issue Stream

Sentry correctly catches the exception in the `internal` project, and it seems to be caused in the `POST /extensions/bitbucket/installed/` call by the following access to the BitBucket payload:

https://github.com/getsentry/sentry/blob/8dc4fc12370ad1eb92f27dd874cdb9e124f41d5f/src/sentry/integrations/bitbucket/integration.py#L176

On master the line is slightly different:

https://github.com/getsentry/sentry/blob/fb0d4732215ff8471c461aaf3729ace4f2f5bb86/src/sentry/integrations/bitbucket/integration.py#L169

As Sentry reports, the error is caused by the `username` key of `principal_data` actually being `nickname`:

```python
{
	account_status : u'active',
	created_on : u'redacted',
	display_name : u'redacted',
	has_2fa_enabled : None,
	is_staff : False,
	links : {
		avatar : {
		},
		html : {
		},
		self : {
		}
	},
	nickname : u'redacted username',
	properties : {
	},
	type : u'user',
	uuid : u'{redacted}'
}
```

The change might be due to https://developer.atlassian.com/cloud/bitbucket/bitbucket-api-changes-gdpr/#introducing-atlassian-account-id-and-nicknames (Apr. 29, 2019)

### What you expected to happen

Integration should have succeeded.

### Possible Solution

The attribute to get the username should be `nickname`.

**Comment by mmospanenko** [comment]
Got the same problem, trying to install Master branch but it doesn't work (ImportError: No module named sentry.utils.distutils while installing via pip).

**Comment by BYK** [comment]
Please see this forum answer: https://forum.sentry.io/t/setup-bitbucket-integration/7791/7?u=byk

This issue is fixed in #13779 which didn't make it into the 9.1.2 release. You can use the latest master via our [onpremise repo](https://github.com/getsentry/onpremise).
